bridge/client/app/_components/ktb-stage-overview/ktb-stage-overview.component.ts

```diff
@@ -52,6 +57,17 @@ export class KtbStageOverviewComponent implements OnDestroy, OnInit {
     });
   }
 
+  public ngAfterContentInit(): void {
+    let stageName = this.route.snapshot.queryParamMap.get('stage');
```

Comment:
The query params should be the **single source of truth**. Now, we need to do the exact same steps twice in `ngAfterContentInit` and in `selectStage` (set the selected stage and emit the change).

We can subscribe to the `queryParamMap` here (instead of using the snapshot) and then do the rest of this method inside of the `subscribe` block. In the `selectStage` method we would only trigger the router navigation (and stop the propagation), and let the subscription do the magic.

Reply:
That's actually exactly how I did it at first but then I couldn't propagate the `filterType` from the `selectStage` method.